### PR TITLE
fix: RDF export error for lexical value null

### PR DESCRIPTION
**Pull request by joerivandervelde**
Fix for RDF export bug: after saving a STRING_ARRAY with one or multiple null values, RDF export fails with this exception:

```RDF export failed due to an exception: lexicalValue may not be null```

This null check prevents the problem and lets RDF export succeed.

### CODE REVIEW

File: backend/molgenis-emx2-semantics/src/main/java/org/molgenis/emx2/semantics/rdf/ValueToRDF.java [hunk 1]
```diff
@@ -135,6 +135,9 @@ public static Value applyFormatting(Object o, Column column, String schemaContex
 
     } else {
 
+      if (o == null) {
```
Comment (jelmerveen) [inline backend/molgenis-emx2-semantics/src/main/java/org/molgenis/emx2/semantics/rdf/ValueToRDF.java:138]: I would either connect it to the others in the big if, else if.

But better would it to rewrite as a switch case. (which is performs faster)

**Review by jelmerveen — CHANGES_REQUESTED** [review]

File: backend/molgenis-emx2-semantics/src/main/java/org/molgenis/emx2/semantics/rdf/ValueToRDF.java (hunk 1 above)
Comment (jelmerveen) [reply backend/molgenis-emx2-semantics/src/main/java/org/molgenis/emx2/semantics/rdf/ValueToRDF.java:138]: https://stackoverflow.com/questions/33051931/if-else-vs-switch-performance-in-java

tldr;

Switch perf is better than if else as in case of switch there will be one time evaluation . Once it evaluated the switch it knows which case needs to be executed but in case of if else it has to go through all conditions in case of worst scenario.

The longer the list condition, better will be switch performance but for shorter list (just two conditions), it can be slower also

Comment (joerivandervelde) [reply backend/molgenis-emx2-semantics/src/main/java/org/molgenis/emx2/semantics/rdf/ValueToRDF.java:139]: Great suggestion! This code is heavily used during RDF export so performance matters here. Now implemented as switch.

**Review by jelmerveen — APPROVED** [review]
